### PR TITLE
 implementation ( Content Analytics Search):  #30230 CA search: Add Route and create Angular Library 

### LINKED ISSUE #30230: CA search: Add Route and create Angular Library
### User Story
Add the route and library to be able to use search for Content Analytics.

### Acceptance Criteria

- The Content Analytics Portlet should be accessible from `/dotAdmin/#/content-analytics`
- Angular Library should be ready to be implemented
- 
- . 
### dotCMS Version
master

### Proposed Objective
_to_define_

### Proposed Priority
_to_define_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
_to_define_

### Quality Assurance Notes & Workarounds
_to_define_

### Sub-Tasks & Estimates

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Enable the portlet for content analytics search
* Create the angular library. 

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

**Review by fmontes — APPROVED** [review]

File: core-web/libs/portlets/dot-content-analytics/portlet/src/lib/lib.routes.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import { DotContentAnalyticsComponent } from './dot-content-analytics/dot-content-analytics.component';
+
+export const DotContentAnalyticsRoutes: Route[] = [
+    {
+        path: '',
+        component: DotContentAnalyticsComponent,
+        providers: [DotPushPublishEnvironmentsResolver, DotEnterpriseLicenseResolver],
+        resolve: {
+            pushPublishEnvironments: DotPushPublishEnvironmentsResolver,
```
Comment (oidacra) [inline core-web/libs/portlets/dot-content-analytics/portlet/src/lib/lib.routes.ts:13]: This portlet use pushPublish?

File: core-web/apps/dotcms-ui/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -70,6 +70,18 @@ const PORTLETS_ANGULAR: Route[] = [
...
             import('@dotcms/portlets/dot-locales/portlet').then((m) => m.DotLocalesRoutes)
     },
+    {
+        path: 'content-analytics',
+        canActivate: [MenuGuardService],
+        canActivateChild: [MenuGuardService],
+        data: {
+            reuseRoute: false
+        },
+        loadChildren: () =>
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/app-routing.module.ts:80]: maybe later change it to `loadComponent`, this route never will have child routes.

File: core-web/libs/portlets/dot-content-analytics/portlet/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# dot-content-analytics
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `nx test dot-content-analytics` to execute the unit tests.
```
Comment (oidacra) [inline core-web/libs/portlets/dot-content-analytics/portlet/README.md:7]: Add what this portlet is. It's like a small doc.

File: core-web/libs/portlets/dot-content-analytics/portlet/project.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "name": "dot-content-analytics",
+    "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
+    "sourceRoot": "libs/portlets/dot-content-analytics/portlet/src",
+    "prefix": "lib",
+    "projectType": "library",
+    "tags": [],
```
Comment (oidacra) [inline core-web/libs/portlets/dot-content-analytics/portlet/project.json:7]: add here  "tags": ["type:feature", "scope:dotcms-ui", "portlet:dev-content-analytic"]

File: core-web/libs/portlets/dot-content-analytics/portlet/src/lib/lib.routes.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-content-analytics/portlet/src/lib/lib.routes.ts:13]: maybe not, will be removed if is the case.

**Review by zJaaal — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/libs/portlets/dot-content-analytics/portlet/project.json (hunk 4 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-content-analytics/portlet/project.json:7]: done

File: core-web/apps/dotcms-ui/src/app/app-routing.module.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/app-routing.module.ts:80]: done

**Pull request merged**